### DISCUSSION TITLE
aa_encode.py or SR4_class_recoder.py

### DISCUSSION THREAD

**Discussion by achoubc** [TheBrownLab/PhyloFisher#68] [General]
Hello,
I tried recoding my concatenated matrix using aa_encode,py but for some reason, it spits out 20 duplicates of each taxon in the output file (in v1.2.4). 
The same thing also happened when I used SR4_class_recoder.py in v1.1.2. 

There's one more issue with the v1.2.4 aa_encode.py.. The recoded output looks something like this:
>Taxon1
23404H6009275D3E9A105D142739H5205AADE06019A08DA1B5A1254458BH
72F35BG007114H34D9A14D00D9H58D943H5E10AEA735848H53H0---AF703
5H0755HA053DE099687D822AH4D2117541E085H6A530D05AE5817HD3102H
0556105811D9503553EE591314HAHA50A14D3E4G053250F07B06D5745FHF
83218B45D68283BH480F01A36809BDA2398H454951344127B05EAB60ADH3
954EG55B332FF74H8169F30D4E14D2150BG44977703757B94D023GH5088H
D3B5260F77G308H51ABDBFH035H4409H0H721E0B30F55DA055E94107HHF1
29A8H228774978H20G93A74A3D881D9E849307352HE6H012747H9785D3G9
0F54D38E71E4EA792H739116H485048479977HF0190F514A3HH17H9D20H4
2873717D74F101541991HH1HG98084EAHF37H0437684B50H847G9800D3G5
8405525EHB355030744B48F041546D401H240674D701B015AF2360130875
0HE9883311DBE51140965B31116B4806H18394D000A754D0585794H44215
505D7450FF09A743B13E31739H50HD359601041H2BDA59H4HD33GD5F6AA7
34335DF44F1H33A17A09D5-A031455H2HHE1180810D1E35EDD2605981868
8H743D0A7121931607FF0530736F99A1763DH56---------------------
---11E615574444AF301FE00A5D116F18004B704800340A513FE353G298

The command used was aa_recode.py -i matrix.fas -re D18 but the result was the same with other -re options.
I haven't tried this on the latest 1.2.8v though..

I can email you sub-set of input .fas or the output. Seems I can't attach them here. 
Thank you for your time!
Best,

**Comment by atice** [comment]
Hi Anna,

Thank you for your email. I will need until Monday to look into this thoroughly and get back to you. The taxon duplication is sounds strange. The D18 coding looks as expected though I believe. Will be back in touch early next week if not sooner.

Alex

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Anna Cho ***@***.***>
Sent: Wednesday, January 25, 2023 7:21:08 PM
To: TheBrownLab/PhyloFisher ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [TheBrownLab/PhyloFisher] aa_encode.py or SR4_class_recoder.py (Discussion #68)


Hello,
I tried recoding my concatenated matrix using aa_encode,py but for some reason, it spits out 20 duplicates of each taxon in the output file (in v1.2.4).
The same thing also happened when I used SR4_class_recoder.py in v1.1.2.

There's one more issue with the v1.2.4 aa_encode.py.. The recoded output looks something like this:

Taxon1
23404H6009275D3E9A105D142739H5205AADE06019A08DA1B5A1254458BH
72F35BG007114H34D9A14D00D9H58D943H5E10AEA735848H53H0---AF703
5H0755HA053DE099687D822AH4D2117541E085H6A530D05AE5817HD3102H
0556105811D9503553EE591314HAHA50A14D3E4G053250F07B06D5745FHF
83218B45D68283BH480F01A36809BDA2398H454951344127B05EAB60ADH3
954EG55B332FF74H8169F30D4E14D2150BG44977703757B94D023GH5088H
D3B5260F77G308H51ABDBFH035H4409H0H721E0B30F55DA055E94107HHF1
29A8H228774978H20G93A74A3D881D9E849307352HE6H012747H9785D3G9
0F54D38E71E4EA792H739116H485048479977HF0190F514A3HH17H9D20H4
2873717D74F101541991HH1HG98084EAHF37H0437684B50H847G9800D3G5
8405525EHB355030744B48F041546D401H240674D701B015AF2360130875
0HE9883311DBE51140965B31116B4806H18394D000A754D0585794H44215
505D7450FF09A743B13E31739H50HD359601041H2BDA59H4HD33GD5F6AA7
34335DF44F1H33A17A09D5-A031455H2HHE1180810D1E35EDD2605981868
8H743D0A7121931607FF0530736F99A1763DH56---------------------
---11E615574444AF301FE00A5D116F18004B704800340A513FE353G298

The command used was aa_recode.py -i matrix.fas -re D18 but the result was the same with other -re options.
I haven't tried this on the latest 1.2.8v though..

I can email you sub-set of input .fas or the output. Seems I can't attach them here.
Thank you for your time!
Best,

**Reply by achoubc** [reply]
Thanks Alex! Oh I see that on Hernandez_ryan github and your source,  recoded AA is assigned to numerical values.
I also checked that "-re SR4" option outputs with nucleotide (as it should). My bad! Just to add, there's no output folder with nexus file and fasta file..
